traitlets/traitlets.py

```diff
@@ -872,6 +885,112 @@ def setup_instance(self):
         self._trait_validators = {}
         super(HasTraits, self).setup_instance()
 
+    @classmethod
+    def setup_class(cls):
+        cls._write_docs_to_class()
+        MetaHasTraits.setup_class(cls)
+
+    @classmethod
+    def _write_docs_to_class(cls):
+        if hasattr(cls.__init__, 'im_func'):
+            docs = cls._rst_trait_help() or ""
+            if docs:
+                # copy the image function
+                im = cls.__init__.im_func
+                n, c, d, g = (im.__name__, im.__code__,
+                              im.__defaults__, im.__globals__)
+                f = types.FunctionType(c, g, n, d, im.__closure__)
```

Comment:
On Python 3, you should be able to set `method.__doc__ = 'foo'` without this copying.